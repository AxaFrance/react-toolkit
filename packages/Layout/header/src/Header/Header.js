import React from 'react';
import { ClassManager, Constants } from '@axa-fr/react-toolkit-core';

const propTypes = {
  ...Constants.propTypes,
};

const defaultClassName = 'af-header';
const defaultProps = {
  ...Constants.defaultProps,
  className: defaultClassName,
};

const Header = ({ classModifier, className, children, id }) => {
  const componentClassName = ClassManager.getComponentClassName(
    className,
    classModifier,
    defaultClassName
  );
  return (
    <div className={componentClassName} id={id}>
      <div className="container af-header__wrapper">
        <header className="af-header__content" role="banner">
          {children}
        </header>
      </div>
    </div>
  );
};

Header.propTypes = propTypes;
Header.defaultProps = defaultProps;

export default Header;
