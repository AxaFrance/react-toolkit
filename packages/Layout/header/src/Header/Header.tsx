import React, { ReactNode } from 'react';
import { ClassManager } from '@axa-fr/react-toolkit-core';

type Props = {
  children: ReactNode;
  className?: string;
  classModifier?: string;
};

const Header = (props: Props) => {
  const { classModifier, className, children } = props;
  const componentClassName = ClassManager.getComponentClassName(
    className,
    classModifier,
    'af-header'
  );
  return (
    <div className={componentClassName}>
      <div className="container af-header__wrapper">
        <header className="af-header__content" role="banner">
          {children}
        </header>
      </div>
    </div>
  );
};

export default Header;
