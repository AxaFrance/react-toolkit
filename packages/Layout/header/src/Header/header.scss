@import '@axa-fr/react-toolkit-core/src/common/scss/core.scss';

.af-header {
  &__content {
    padding-bottom: 2rem;
    display: flex;
    justify-content: space-between;
    align-items: center;
    flex-wrap: wrap;
  }
}

@include media-breakpoint-up(sm) {
  .af-header {
    &__wrapper {
      width: 100%;
    }
  }
}
@include media-breakpoint-up(md) {
  .af-header {
    &__wrapper {
      width: nth(nth($container-max-widths, 4), 2);
      max-width: 100%;
    }

    &__content {
      flex-wrap: initial;
    }
  }
}
