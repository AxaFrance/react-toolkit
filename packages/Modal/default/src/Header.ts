import {
  withClickId,
  WithClickIdProps,
  compose,
  withProps,
} from '@axa-fr/react-toolkit-core';
import HeaderCore, { HeaderCoreProps } from './HeaderCore';

const onCancelEvent = 'onCancel';

export type HeaderProps = WithClickIdProps<
  HeaderCoreProps,
  typeof onCancelEvent
> & {
  title: string;
};

const setWithProps = (props: HeaderProps) => ({
  ...props,
  children: props.title,
});

const enchance = compose<HeaderCoreProps, HeaderProps>(
  withClickId({ event: [onCancelEvent] }),
  withProps<HeaderProps, HeaderCoreProps>(setWithProps)
)(HeaderCore);

enchance.displayName = 'Header';

export default enchance;
