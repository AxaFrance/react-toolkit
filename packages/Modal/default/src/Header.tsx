import React, { MouseEventHandler } from 'react';
import { getComponentClassName, getClickId } from '@axa-fr/react-toolkit-core';

export type HeaderProps = React.HTMLAttributes<HTMLDivElement> & {
  className?: string;
  onCancel: MouseEventHandler<HTMLButtonElement>;
  title?: string;
  classModifier?: string;
  closeButtonAriaLabel?: string;
};

const Header = ({
  className,
  classModifier,
  title,
  closeButtonAriaLabel = 'Fermer la boite de dialogue',
  ...props
}: HeaderProps) => {
  const componentClassName = getComponentClassName(
    className,
    classModifier,
    'af-modal__header'
  );

  const propsWithClickId = getClickId<HeaderProps>({
    option: {
      event: ['onCancel'],
    },
    props,
  });

  return (
    <header className={componentClassName} {...props}>
      <h4 className="af-modal__header-title">{title}</h4>
      <button
        className="af-modal__header-close-btn"
        type="button"
        aria-label={closeButtonAriaLabel}
        onClick={propsWithClickId.onCancel}>
        <span className="glyphicon glyphicon-close" aria-hidden="true" />
      </button>
    </header>
  );
};

export default Header;
