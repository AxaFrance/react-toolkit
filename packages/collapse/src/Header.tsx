import React from 'react';
import classNames from 'classnames';

import { ClassManager, Constants } from '@axa-fr/react-toolkit-core';

const defaultClassName = 'af-accordion__item-header';

const defaultProps = {
  ...Constants.defaultProps,
  collapse: true,
  className: defaultClassName,
  href: '#collapseOne',
};

export type HeaderToggleElement = {
  collapse: boolean;
  index: number;
  id: string;
};

export type HeaderProps = Partial<typeof defaultProps & { index: number }> & {
  children: React.ReactNode;
  onToggle?: (element: HeaderToggleElement) => void;
  ariaControls?: string;
  id?: string;
};

const Header = ({
  children,
  collapse,
  className,
  classModifier,
  id,
  href,
  ariaControls,
  index,
  onToggle,
}: HeaderProps) => {
  const onToggleEvent = (e: React.MouseEvent) => {
    e.preventDefault();
    if (onToggle) {
      onToggle({
        collapse: !collapse,
        id,
        index,
      });
    }
  };

  const onKeyToggle = (e: React.KeyboardEvent) => {
    if (e.key === 'ArrowDown' || e.key === 'ArrowUp') {
      e.preventDefault();
      if (onToggle) {
        onToggle({
          collapse: !collapse,
          id,
          index,
        });
      }
    }
  };

  const chevronClassName = classNames('glyphicon', {
    'glyphicon-menu-down': !collapse,
    'glyphicon-menu-up': collapse,
  });

  const componentClassName = ClassManager.getComponentClassName(
    className,
    classModifier,
    defaultClassName
  );

  return (
    <div className={componentClassName} role="tab" id={id}>
      <h3 className="af-accordion__item-title" role="intera">
        <span className={chevronClassName} />
        <a
          className="af-accordion__item-toggle"
          data-toggle="collapse"
          href={href}
          aria-expanded={collapse}
          aria-controls={ariaControls}
          onClick={onToggleEvent}
          onKeyDown={onKeyToggle}>
          {children}
        </a>
      </h3>
    </div>
  );
};

Header.defaultProps = defaultProps;

export default Header;
