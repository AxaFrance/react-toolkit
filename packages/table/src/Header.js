import React from 'react';
import PropTypes from 'prop-types';

import { Constants, ClassManager } from '@axa-fr/react-toolkit-core';

import Tr from './Tr';

const propTypes = {
  ...Constants.propTypes,
  children: PropTypes.any.isRequired,
};

const defaultClassName = 'af-table__thead';
const defaultProps = {
  ...Constants.defaultProps,
  className: defaultClassName,
};

const Header = props => {
  const { children, headers, className, classModifier } = props;
  const componentClassName = ClassManager.getComponentClassName(
    className,
    classModifier,
    defaultClassName
  );
  if (headers) {
    return (
      <thead className={componentClassName}>
        <Tr>{children}</Tr>
      </thead>
    );
  }
  return <thead className="af-table-thead">{children}</thead>;
};

Header.propTypes = propTypes;
Header.defaultProps = defaultProps;
Header.displayName = 'Table.Header';

export default Header;
