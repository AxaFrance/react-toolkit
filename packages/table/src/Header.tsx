import React from 'react';
import {
  withClassDefault,
  withClassModifier,
  WithClassModifierOptions,
  compose,
} from '@axa-fr/react-toolkit-core';

import Tr from './Tr';

const DEFAULT_CLASSNAME = 'af-table__thead';

export type HeaderComponentProps = React.HTMLProps<HTMLTableSectionElement>;

const Header: React.SFC<HeaderComponentProps> = props => {
  const { children, headers, className } = props;

  if (headers) {
    return (
      <thead className={className}>
        <Tr>{children}</Tr>
      </thead>
    );
  }
  return <thead className="af-table-thead">{children}</thead>;
};

export type HeaderProps = HeaderComponentProps & WithClassModifierOptions;
const enhance = compose<HeaderComponentProps, HeaderProps>(
  withClassDefault(DEFAULT_CLASSNAME),
  withClassModifier
);

Header.displayName = 'Table.Header';

export default enhance(Header);
