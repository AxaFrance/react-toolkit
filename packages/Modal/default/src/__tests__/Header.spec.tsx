import React from 'react';
import { render, screen } from '@testing-library/react';
import UserEvent from '@testing-library/user-event';
import Header from '../Header';

describe('<Header>', () => {
  it('should render header', () => {
    const { asFragment } = render(<Header title="title" onCancel={() => {}} />);
    expect(asFragment()).toMatchSnapshot();
    expect(screen.getByText('title')).toBeInTheDocument();
  });

  it('should render id click onCancel', () => {
    const onCancel = jest.fn();
    render(
      <Header title="title" onCancel={onCancel}>
        header
      </Header>
    );
    UserEvent.click(screen.getByRole('button', { name: 'Close' }));
    expect(onCancel).toHaveBeenCalledWith({ id: '' });
  });
});
