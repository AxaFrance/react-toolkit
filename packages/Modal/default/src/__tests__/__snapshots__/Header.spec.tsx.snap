// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Header> should render header 1`] = `
<DocumentFragment>
  <div
    class="af-modal__header"
  >
    <h4
      class="af-modal__header-title"
    >
      title
    </h4>
    <button
      aria-label="Close"
      class="af-modal__header-close-btn"
      type="button"
    >
      <span
        aria-hidden="true"
        class="glyphicon glyphicon-close"
      />
    </button>
  </div>
</DocumentFragment>
`;
